glmnet/errors.py

```diff
@@ -0,0 +1,43 @@
+import warnings
+
+
+def _check_glmnet_error_flag(jerr, n_lambda):
+    """Check the error flag. Issue warning on convergence errors (jerr < 0)
+    and exception on anything else."""
+
+    if jerr == 0:
+        return
+
+    if jerr > 0:
+        _fatal_errors(jerr, n_lambda)
+
+    if jerr < 0:
+        _convergence_errors(jerr, n_lambda)
+
+
+def _fatal_errors(jerr, n_lambda):
```

Comment:
Is there a reason to pass `n_lambda` here? Looks like the function doesn't refer to it

Reply:
No, it's only there to make this function consistent with `_convergence_errors`. Although it looks like it may be necessary to decode the additional errors that could be returned by `lognet`.

Reply:
Hmm... looks like `n_lambda` isn't needed in either case. The convergence code is in [-k, -1].